### PR TITLE
Make dummy dummy resolve deps

### PR DESCRIPTION
**Pull request by pforai**
I find this to be more intuitive and it works they way you expect EB to work. 
@fgeorgatos, @pescobar, @wpoely86 opinions?

### CODE REVIEW

**Comment by wpoely86** [comment]
@boegel what is the historical reason that `dummy-dummy` doesn't load deps?

File: easybuild/tools/toolchain/toolchain.py [hunk 1]
```diff
@@ -333,10 +333,13 @@ def prepare(self, onlymod=None):
         if not self._toolchain_exists():
             self.log.error("No module found for toolchain: %s" % self.mod_short_name)
 
-        if self.name == DUMMY_TOOLCHAIN_NAME:
+        
+	if self.name == DUMMY_TOOLCHAIN_NAME:
```
Comment (stdweird) [inline easybuild/tools/toolchain/toolchain.py]: spaces, no tabs for indentation

File: easybuild/tools/toolchain/toolchain.py [hunk 2]
```diff
@@ -333,10 +333,13 @@ def prepare(self, onlymod=None):
...
             if self.version == DUMMY_TOOLCHAIN_VERSION:
                 self.log.info('prepare: toolchain dummy mode, dummy version; not loading dependencies')
-            else:
+		#@todo I keep this is as, but want dummy to do the same as a regular toolchain. Need to think this through.
```
Comment (stdweird) [inline easybuild/tools/toolchain/toolchain.py]: at the very least, also cleanup the log message.
wrt the todo, any progress? i think this requires an extra option to have this behaviour. (as it's very backwards incompatible).

**Comment by stdweird** [comment]
@wpoely86 i think that's the only difference between `dummy` and `dummy-dummy`. there must be some docs about this somewhere

**Comment by wpoely86** [comment]
@stdweird as fars as I know, it's the only difference (it's in the readthedocs). But why did you make this difference in the first place? There must have been some reason?

**Comment by stdweird** [comment]
@wpoely86 don't remember immediately, but one thing @boegel is going to add are runtime deps, which is basicallly what the original `dummy-dummy` hack is all about.

**Comment by stdweird** [comment]
@pforai do you need `dummy-dummy` or were you not aware of this difference?

**Comment by boegel** [comment]
The only reason why `dummy` with `dummy` version doesn't load deps is because it has been a crippled toolchain from the very start. The idea behind `dummy` was that it should only be used to build compilers with, to kick of a toolchain, and then build everything else with a proper toolchain (which is potentially compiler-only).

That's not how people have been using it however. There has been a call for proper support for building with the system-provided tools, so we should have a proper `system` toolchain instead (cfr. #1181).

So, we should:

* deprecate the usage of `dummy` (both with and without `dummy` version)
* support a `system` toolchain to replace it (where we don't have this 'don't load deps' trickery)
* switch from using `dummy` to `system` in all easyconfigs where needed, and then simply remove `dummy` in EB v3.0

**Comment by pforai** [comment]
@boegel agreed. I guess providing a SystemGCC toolchain is quite easy to do. But I would make dummy dummy behave as all the others until it is removed in 3.0.

**Comment by boegel** [comment]
@pforai: not a big fan of modifying behaviour of something from one minor version to another, especially if we're going to deprecate it...

Shouldn't be too much work to deprecate `dummy`.

**Pull request closed without merging**